### DISCUSSION TITLE
Google Oauth scope issues

### DISCUSSION THREAD

**Discussion by starascendin** [pocketbase/pocketbase#4321] [Q&A]
Hi friends, I am running into some issues w/ integrating w/ Google Oauth provider w/ the specific scopes. The code is below:

```
# login
  const authResponse = await authCollection.authWithOAuth2({
    provider: provider.name,
    createData: {
      // emailVisibility: true,
    },
    scopes: ["https://www.googleapis.com/auth/calendar", "https://www.googleapis.com/auth/youtube.readonly"]
  });
  console.log("#authResponse", authResponse)

```
I get the 401 response in the localhost when i add scopes, but when i remove the scopes I can proceed w/ oauth flow fine.
Am i doing something wrong here in terms of wanting to define the scopes? In the end, I want to get the access token / refresh token so I can interface w/ google apis

```
# response in localhost
2024/02/09 10:28:51 failed to fetch OAuth2 user profile via https://www.googleapis.com/oauth2/v1/userinfo (401):
{
  "error": {
    "code": 401,
    "message": "Request is missing required authentication credential. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.",
    "status": "UNAUTHENTICATED"
  }
}


```

**Comment by ganigeorgiev** [comment]
I'm not familiar with your used scopes but I think you'll need at least the `userinfo.profile` scope (and `userinfo.email` if you want the email).

You can try to add the below to your scopes:

```
"https://www.googleapis.com/auth/userinfo.profile",
"https://www.googleapis.com/auth/userinfo.email",
```

**Reply by starascendin** [reply]
oops got it, my silly mistake for missing the 2 required scopes. Thanks!
